src/Core/Checkout/Customer/Exception/InvalidLoginAsCustomerTokenException.php

```diff
@@ -0,0 +1,27 @@
+<?php declare(strict_types=1);
+
+namespace Shopware\Core\Checkout\Customer\Exception;
+
+use Shopware\Core\Framework\ShopwareHttpException;
+use Symfony\Component\HttpFoundation\Response;
+
+class InvalidLoginAsCustomerTokenException extends ShopwareHttpException
```

Comment:
Please see https://github.com/shopware/platform/blob/trunk/adr/2022-02-24-domain-exceptions.md

Reply:
@shyim What would the best possible here?

As an example `Shopware\Core\Checkout\Customer\Exception\CustomerAuthThrottledException` looks same to my exception class and is also handled by try catch in few controllers.

I could also introduce the exception factory in `Shopware\Core\Checkout\Customer\CustomerException`, or just removing try catch statements in `AuthContoller::loginAsCustomer` would also make sense i think.

Reply:
We want to make any new exception using an exception factory. The existing ones will be changed in the next major